### PR TITLE
Got rid of the specialized `solve`

### PR DESCRIPTION
**Pull request by pwl**
Removed the `solve` specialization and replaced it with the constructor specialization

### CODE REVIEW

File: src/dense.jl [hunk 1]
```diff
@@ -50,18 +50,17 @@ end
...
-                    ::Type{DenseOutput{I}};
-                    opts...)
+@compat function (::Type{DenseOutput{I}}){T,I}(ivp::IVP{T};
+                                               opts...)
     # create integrator
-    integ = I{T}(; opts...)
+    integ = I(ivp; opts...)
     # TODO: a nasty workaround: this triggers an error if the method
     # is not registered supported, we ignore the output
     solver = solve(ivp,I;opts...)
```
Comment (pwl) [inline src/dense.jl:59]: this is no longer necessary

**Comment by mauro3** [comment]
I'll do a counter PR...

**Comment by mauro3** [comment]
Actually, I don't have time and you're right (edited).  But how about we get rid of `solve`?  And just use the constructors (be it of the solver or the problem).

File: src/dense.jl (hunk 1 above)
Comment (mauro3) [inline src/dense.jl:59]: `solver = solve(ivp,I;opts...)` neither because above should error, no?

**Comment by mauro3** [comment]
Add a constructor to `Problem`:

```
Problem{S<:AbstactSolver}(ivp::IVP, ::Type{S}; opts...) = Problem(ivp, S(ivp; opts...))
```

**Pull request merged**